tests/virt_autotest/login_console.pm

```diff
@@ -43,6 +43,26 @@ sub login_to_console {
     if (check_var('PERF_KERNEL', '1') or check_var('CPU_BUGS', '1') or check_var('VT_PERF', '1')) {
         reset_consoles;
         select_console 'sol', await_console => 0;
+        if (get_var("XEN_BOOT")) {
+            assert_screen 'pxe-qa-net-mitigation', 90;
+            send_key 'ret';
+            send_key 'up';
+            send_key_until_needlematch [qw(grub2 grub1)], 'up';
+            send_key 'up';
+            reset_consoles;
+            select_console 'sol', await_console => 0;
+            send_key 'up';
+            save_screenshot;
+            for (1 .. 20) {
+                if ($_ == 10) {
+                    reset_consoles;
+                    select_console 'sol', await_console => 0;
+                }
+                send_key 'down';
+                last if check_screen 'virttest-bootmenu-xen-kernel', 5;
+            }
+            return;
+        }
```

Comment:
IMHO, you only need to have special treatment for "   assert_screen 'pxe-qa-net-mitigation', 90;" , all other logic of the added lines can work well in existing code logic(overall in this function, but not only in this if ).

Reply:
Thanks, update login_console.pm as your suggestion.